src/ophys_etl/transforms/roi_transforms.py

```diff
@@ -127,3 +129,149 @@ def crop_roi_mask(roi_mask: coo_matrix) -> coo_matrix:
+        The max motion correction values identified in the motion correction
+        step of ophys segmentation pipeline
+    movie_shape: Tuple[int, int]
+        The frame shape of the movie from which ROIs were extracted in order
+        of: (height, width).
+
+    Returns
+    -------
+    List[Dict[str, Any]]
+        A list of dictionaries representing the ROIs in old segmentation
+        format, the data contained inside each one is as follows:
+        {
+            id: int
+            x: int
+            y: int
+            width: int
+            height: int
+            valid_roi: Bool
+            mask_matrix: List[List[Bool]]
+            max_correction_up: int
+            max_correction_down: int
+            max_correction_left: int
+            max_correction_right: int
+            mask_image_plane: int
+            exclusion_labels: List[int] codes are defined in 2019 Ophys Docs
+        }
+        For details about specific values see design document for 2020 Ophys
+        Segmentation Refactor and Update
+        """
+    old_rois = []
+    for temp_id, coo_mask in enumerate(coo_masks):
+        old_roi = _coo_mask_to_old_format(coo_mask)
+        old_roi['id'] = temp_id  # popped off writing to LIMs
+        old_roi['cell_specimen_id'] = temp_id  # updated post nway cellmatching
+        old_roi['max_correction_up'] = max_correction_vals.up
+        old_roi['max_correction_down'] = max_correction_vals.down
+        old_roi['max_correction_right'] = max_correction_vals.right
+        old_roi['max_correction_left'] = max_correction_vals.left
+        old_roi['mask_image_plane'] = 0
+        old_roi['exclusion_labels'], old_roi['valid_roi'] = _check_exclusion(
+            old_roi, movie_shape)
+        old_rois.append(old_roi)
+    return old_rois
+
+
+def _coo_mask_to_old_format(coo_mask: coo_matrix) -> Dict:
+    """
+    This functions transforms ROI mask data from COO format
+    to the old format used in older segmentation. This function
+    writes only the data associated with the mask.
+    Parameters
+    ----------
+    coo_mask: coo_matrix
+        The coo roi matrix to be converted to old segmentation mask format
+
+    Returns
+    -------
+    Dict:
+        A dictionary that contains the mask data from the coo matrix in the
+        old segmentation format
+        {
+            'x': int (x location of upper left corner of roi in pixels)
+            'y': int (y location of upper left corner of roi in pixels)
+            'width': int (width of the roi mask in pixels)
+            'height': int (height of the roi mask in pixels)
+            'mask_matrix': List[List[bool]] (dense matrix of roi mask)
+        }
+    """
+    bounds = roi_bounds(coo_mask)
+    height = bounds[1] - bounds[0]
+    width = bounds[3] - bounds[2]
+    mask_matrix = crop_roi_mask(coo_mask).toarray()
+    mask_matrix = np.array(mask_matrix, dtype=bool)
+    old_roi = {
+        'x': int(bounds[0]),
```

Comment:
Is this cast to `int` necessary? I thought `bounds` should already be returning ints?

Reply:
They come out of the bounds function as np int64 which are not serializable.